src/package.lisp

```diff
@@ -324,6 +324,8 @@
    #:codegen-program
    #:compile-expression
    #:update-function-env
+   #:a1 #:a2 #:a3 #:a4 #:a5 #:a6 #:a7 #:a8 #:a9 #:a10
```

Comment:
I'm not sure if this helps but there are functions `construct-function-entry` and `apply-function-entry` which could be wrapped with a macro to automatically select the correct function. I'm fine exposing the direct functions too but I think we should discuss what we want the external interface to be for this.